contents/blog/how-we-turned-clickhouse-into-our-eventmansion.md

```diff
@@ -0,0 +1,84 @@
+
+- *Real-time results*: Our users want PostHog to stay as fast as it was in the beginning. ClickHouse’s performance improvements were such that, with a little cleverness in the way we architect, we’re still able to provide answers in real-time.
+- *Future-proofing:* We need to constantly grow our feature set. ClickHouse SQL was developed to resemble Postgres and MySQL syntax, which means we’ve been able to continue adding new features without ClickHouse becoming a bottleneck.
+- *Filter on user-defined properties:* We leverage ClickHouse’s JSON deserialization functionality and features such as [materialized columns](https://posthog.com/blog/clickhouse-materialized-columns) to provide this, albeit at the cost of some CPU overhead.
+
+## Making the switch
+
+Normally, the process of switching to a new database model could be quite tricky. Luckily however, we’d just created feature flags within PostHog and were able to leverage this when switching. 
+
+First, we deployed ClickHouse in parallel to Postgres and begane ingesting data simultaneously into both. We then put all of our features and internal tools behind a feature flag in PostHog, reimplemented our analytic queries one by one and then used the feature flags to switch over. 
+
+Once everything was migrated we were able to simply stop storing data on Postgres while our users began deploying PostHog through our Helm Chart, which we use because of the amazing [ClickHouse-Operator made by Altinity](https://altinity.com/kubernetes-operator/). Easy. 
+
+## The future of ClickHouse on PostHog
+
+It’s fair to say that we bet the farm on ClickHouse, but it’s definitely turned out to be a good bet so far. ClickHouse has recently [raised a significant amount of funding](https://uk.finance.yahoo.com/news/clickhouse-raises-250m-series-b-140000178.html) to secure its long-term future and accelerate its product development, but more importantly it’s created a better experience for PostHog users. 
+
+So, naturally we’re thinking about how to do even more with ClickHouse and build deeper connections with the technology. 
+
+One of the things we’re interested in exploring, for example, is the possibility of building functions at the source instead of writing queries for the functionality. Our hope is that adding these functions to a master code base will be more efficient and we won’t need to rely on SQL to do so much heavy lifting. We’re looking to [hire a C++ Engineer](https://posthog.com/careers) who can help us achieve this. 
+
+We’re also currently experimenting with MaterializedPostgresSQL engine as a way to consume the logical representation log coming from Postgres and materializes it in a mergetree table on ClickHouse. Data drift is a huge pain point for any near-data product and we’re hopeful we can find a way to guarantee to some extent the quality of data in ClickHouse compared to our source of truth in Postgres. 
```

Comment:
Google suggests there isn't something called "MaterializedPostgresSQL" and these words should be split out.

We should link to a relevant resource too.